.github/workflows/cicd-pipeline.yml

```diff
@@ -14,44 +14,62 @@ jobs:
     - uses: actions/checkout@v2
 
     - name: Set up Python 3.9
+      id: setup-python
       uses: actions/setup-python@v2
       with:
         python-version: 3.9
 
-    - name: Install pipenv
+    - name: Install build requirements
       run: |
         python -m pip install --upgrade pip
-        pip install pipenv
+        pip install pipenv invoke
       shell: bash
 
-    - name: Cache dependencies
-      id: cache-deps-webapp
-      uses: actions/cache@v1
+    - name: Python dependencies - configure cache
+      uses: actions/cache@v2
       with:
         path: ~/.local/share/virtualenvs
-        key: ${{ runner.os }}-deps-webapp-v${{ secrets.CACHE_VERSION }}-${{ hashFiles('webapp/Pipfile.lock') }}
+        key: ${{ runner.os }}-webapp-v${{ secrets.CACHE_VERSION }}-python-${{ steps.setup-python.outputs.python-version }}-pipenv-${{ hashFiles('webapp/Pipfile.lock') }}
```

Comment:
I picked that up from the documented examples for that action and it makes sense: if we switch Python versions for some reason, we shouldn't use an old cache. (Not very important, since that build would likely just fail and thus invalidate the cache, but hey…)

Reply:
Yes that makes sense